src/js/uri_scheme.js

```diff
@@ -0,0 +1,75 @@
+import $ from 'jquery';
+
+import GUI from './gui';
+import serial from './serial';
+import PortHandler from './port_handler';
+import CONFIGURATOR from './data_storage.js';
+
+export async function handleURIScheme() {
+    console.log(`Listening for ${CONFIGURATOR.uriScheme}:// URI scheme invocations/events`);
+
+    if (GUI.isNWJS()) {
+        handleNWJSArgv();
+        handleNWJSOpenEvents();
+    } else if (GUI.isCordova()) {
+        handleCordovaOpenEvents();
+    }
+}
+
+function connectTcp(connectionString, eventType) {
+    if (!connectionString) {
+        return;
+    }
+
+    console.log(`Processing ${eventType} with connection string "${connectionString}"`);
+
+    const protocolScheme = `${CONFIGURATOR.uriScheme}://`;
+    // NOTE: On some platforms a trailing slash is added automatically (it needs to be excluded)
+    const connectionStringRegex = new RegExp(`^.*${protocolScheme}(.+?)/?$`);
+
+    const url = connectionString.match(connectionStringRegex)?.[1];
+    if (!url) {
+        console.log(`Invalid ${protocolScheme} connection string "${connectionString}" from ${eventType}`);
+        return;
+    }
+
+    const tcpUrl = `tcp://${url}`;
+    if (!serial.tcpUrlRegex.exec(tcpUrl)) {
+        console.log(`Invalid ${protocolScheme} TCP URL ${tcpUrl} (from ${eventType} with connection string "${connectionString}")`);
+        return;
+    }
+
+    console.log(`Connecting to ${tcpUrl} (from ${eventType} with connection string "${connectionString}")`);
+
+    // Explicitly add the 'manual' select option if the port options haven't loaded yet
+    // (e.g. during a cold start)
+    if ($('#port [value="manual"]').length === 0){
+        PortHandler.addManualPortSelectOption();
+    }
+
+    $('#port').val('manual').trigger('change');
+    $('#port-override').val(tcpUrl).trigger('change');
+    $('div.connect_controls a.connect').trigger('click');
+}
+
+
+function handleNWJSArgv() {
+    const connectionString = nw?.App?.argv?.[0] || '';
```

Comment:
in what case is this passed in used? Can you provide example usage?

Just wondering if it won't clash with something like `--inspect` oar other debug flags for example.

Reply:
@chmelevskij `nw.App.argv` will be present every time the configurator is _cold_ started with any CLI argument. Thus cold starts through the `btfltcp://` URI scheme end up triggering invocations like `./betaflight-configurator btfltcp://ip` and  `nw.App.argv` gets set to `['btfltcp://...']`. [Here are the NW.js docs on `argv`](https://docs.nwjs.io/en/latest/References/App/#appargv) (it seems some of the NW.js-specific arguments such as `--nwapp` are already filtered out by default).

However you are absolutely right that we might end up running into edge-cases if the user modifies their default execution command (in the `.desktop`/plist/registry). A manual invocation (on Linux) with e.g. `/opt/betaflight/betaflight-configurator/betaflight-configurator --disable-gpu --someTestArgument btltcp://10.0.0.1` results in a `nw.App.argv` value of `['--disable-gpu', '--someTestArgument', 'btltcp://10.0.0.1']` (thus the newly added functionality breaks; but there is no impact for NW.js). 

I think the easiest solution might be to always get the last element in the array instead of the first one - thus:

```suggestion
    const connectionString = (nw?.App?.argv? || []).at(-1);
```
(since the URI scheme always seems to be added at the end of the array)

I will test that across the platforms and update the MR - thank you! :)

Reply:
Re-tested and updated the MR - the aforementioned change seems to work as expected on all platforms.

Reply:
Can we make it named argument then instead? That would de-couple it from any issues all together. Something like `--bf-uri` or similar? we already do something similar for the build parts https://github.com/betaflight/betaflight-configurator/blob/master/gulpfile.js#L146

Reply:
@chmelevskij Unfortunately from what I checked, NW.js won't let us name the argument (I guess they intended it to be more or less hidden - as it is just appended to the end of the execution string). 

---

However indeed we could add a separate dedicated `--tcp-url={{IP/URL}}` argument for anyone who might want to launch the configurator from the CLI. Should we maybe open a separate issue for expanding its CLI interface? That way we can use the opportunity to also add e.g. a help menu and a few more arguments. I can imagine that this might be useful for power users/manufacturers. For example some rudimentary functionality for backup management might be nice - such as: 
- `./betaflight-configurator --tcp-url=... --diff-all > backup.txt` (establish the connection and pipe the `diff all` CLI output)
- `cat backup.txt | ./betaflight-configurator --tcp-url=... --load-diff` (execute the input from a pipe in the CLI)

Implementation-wise - NW.js also allows us to execute a script in a "pure" Node.js context via [the `node-main` script script in the manifest](https://docs.nwjs.io/en/latest/References/Manifest%20Format/#node-main). There is actually a very basic implementation of `node-main` in [a past revision of this MR](https://github.com/betaflight/betaflight-configurator/compare/897a64f9f744a9e8fc409573b0aac32e5e4fed6f..13deeb1075cefaf09f07b76ae26a70681fcdb9f9) (I tried it because of an external dependency for the URI schema management which I later discarded).